### DISCUSSION TITLE
adding-validation-to-strongly-typed-configuration-objects-using-flentvalidation/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#59] [Announcements]
# adding-validation-to-strongly-typed-configuration-objects-using-flentvalidation/

In this post I describe how to to verify your strongly typed configuration objects are correctly bound to your config at start up using FluentValidation

https://andrewlock.net/adding-validation-to-strongly-typed-configuration-objects-using-flentvalidation/

**Comment by thegrahamking** [comment]
Your putting it all together never calls `ValidateFluentValidation`.

Should 
``` csharp
builder.Services.AddOptions<SlackApiSettings>()
    .BindConfiguration("SlackApi") // 👈 Bind the SlackApi section in config
    .ValidateDataAnnotations() // 👈 Enable validation
    .ValidateOnStart(); // 👈 Validate on app start
```
be
``` csharp
builder.Services.AddOptions<SlackApiSettings>()
    .BindConfiguration("SlackApi") // 👈 Bind the SlackApi section in config
    .ValidateFluentValidation() // 👈 Enable fluent validation
    .ValidateOnStart(); // 👈 Validate on app start
```
?

**Reply by davidhenley** [reply]
This is what I'm wondering too. How does calling ValidateDataAnnotations and ValidateOnStart tie in with Fluent Validation?

**Reply by Van-Dame** [reply]
It should be latter for sure. `ValidateDataAnnotations` is for validating data annotations.

**Reply by ILya-Lev** [reply]
com'on guys, it's just a misprint

**Reply by thegrahamking** [reply]
Exactly, that's all I was trying to point out.

**Comment by RehanSaeed** [comment]
This is good enough that it should be part of [FluentValidation.AspNetCore](https://github.com/FluentValidation/FluentValidation.AspNetCore).

**Comment by ILya-Lev** [comment]
How can we extend ValidationHostedService, so that it writes logs into injected ILogger instead of just throwing an exception?

**Reply by ILya-Lev** [reply]
on the other hand we can inject ILogger<FluentValidationOptions<TOptions>> into the adapter (i.e. FluentValidationOptions) and log from there...

**Comment by chrisxfire** [comment]
In this example, is it still necessary to...
```
builder.Services.AddSingleton(resolver => 
        resolver.GetRequiredService<IOptions<SlackApiSettings>>().Value);
```
...in order to avoid the `IOptions` depedency?
